internal/controller/environments/environments.go

```diff
@@ -123,25 +132,27 @@ func SetupReconcilerWithManager(
 		ctx,
 		&api.Environment{},
 		envsByAppIndexField,
-		func(obj client.Object) []string {
-			env := obj.(*api.Environment) // nolint: forcetypeassert
-			if env.Spec.HealthChecks == nil {
-				return nil
-			}
-			apps := make([]string, len(env.Spec.HealthChecks.ArgoCDAppChecks))
-			for i, appCheck := range env.Spec.HealthChecks.ArgoCDAppChecks {
-				apps[i] =
-					fmt.Sprintf("%s:%s", appCheck.AppNamespace, appCheck.AppName)
-			}
-			return apps
-		},
+		indexEnvsByApp,
```

Comment:
This anonymous in-line function just got extracted into a named function that can be tested.

It's mostly unrelated to this PR except that this was the approach I took for the _new_ index, so I thought I'd apply it here also for consistency.